### PR TITLE
fix batch requests with ws

### PR DESCRIPTION
**Pull request by baabeetaa**
## Description

Batch Request is in spec of jsonrpc https://www.jsonrpc.org/specification#batch

This PR is a fix to make Batch Request working for websocket.

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: rpc/websockets.go [hunk 1]
```diff
@@ -186,100 +186,106 @@ func (s *websocketsServer) readLoop(wsConn *wsConn) {
...
+				if len(params) == 0 {
+					s.sendErrResponse(wsConn, "empty parameters")
+					continue
+				}
+
+				id, ok := params[0].(string)
+				if !ok {
+					s.sendErrResponse(wsConn, "invalid parameters")
+					continue
+				}
```
Comment (fedekunze) [inline rpc/websockets.go:266]: this looks duplicated from `eth_subscribe`. Can you create a function instead?

File: rpc/websockets.go [hunk 2]
```diff
@@ -186,100 +186,106 @@ func (s *websocketsServer) readLoop(wsConn *wsConn) {
...
-
-			if len(params) == 0 {
-				s.sendErrResponse(wsConn, "empty parameters")
-				continue
-			}
-
-			subID := rpc.NewID()
-			unsubFn, err := s.api.subscribe(wsConn, subID, params)
-			if err != nil {
+		} else {
```
Comment (fedekunze) [inline rpc/websockets.go:193]: nit, use `continue` in the `if` above and drop the `else` here

**Review by fedekunze — CHANGES_REQUESTED** [review]
Minor comments, can you also add tests?

**Comment by baabeetaa** [comment]
> Minor comments, can you also add tests?

do you mean websocket tests?

File: rpc/websockets.go (hunk 2 above)
Comment (baabeetaa) [reply rpc/websockets.go]: thanks, much cleaner

**Review by fedekunze — APPROVED** [review]
ACK, would be great to add websocket tests

**Comment by fedekunze** [comment]
@baabeetaa can you update the PR?

**Review by adisaran64 — APPROVED** [review]
@baabeetaa will add tests with [this ticket](https://github.com/evmos/ethermint/issues/1241), can update and merge this PR when possible

**Comment by baabeetaa** [comment]
i'll close this and update changes to https://github.com/evmos/ethermint/pull/1342 with tests.

**Pull request closed without merging**